sw/device/tests/dif/dif_hmac_sanitytest.c

```diff
@@ -14,9 +14,24 @@
 
 #define MAX_FIFO_FILL 10
 
-const test_config_t kTestConfig = {
-    .can_clobber_uart = false,
+const test_config_t kTestConfig;
+
+// This test needs to understand the byte order of the data in the string and
+// the digest values below, as they are laid out for the current processor.
+// RISC-V is little-endian, so the first of these `kHmacTransactionConfig`
+// values is the one we expect to be used, but we include the other for
+// completeness.
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
```

Comment:
I don't love the macro test here just to figure out the endianness. I feel like we should have two tests, and each construct the data in place in the correct endianness. uint8_t[] of correctly-endian'd bytes.

Reply:
I don't think the sanity test should be that comprehensive, remember we're aiming for something that executes the main datapath from software to peripheral to software again quickly.

Reply:
I would personally just `_Static_assert` on the byte order if that's what's desired. Otherwise, you have different code under test depending on the device, which to me feels a little sketchy.

Reply:
Done.